starknet_py/serialization/factory.py

```diff
@@ -148,13 +149,16 @@ def serializer_for_outputs(payload: List[CairoType]) -> OutputSerializer:
     )
 
 
-def serializer_for_event(event: Abi.Event) -> PayloadSerializer:
+def serializer_for_event(event: Abi.Event | EventType) -> PayloadSerializer:
```

Comment:
What about `Event` type for V1 Abi? I assume we can have the same issue that we encountered with V2.

Reply:
In the context of V1 abi, the event has `Abi.Event` type, the same as v0.  I think I have to add tests also for V1

Reply:
The names are a bit tricky here. As I understand, we have two `Abi.Event` types in different modules, corresponding to V0 and V1 respectively.

Reply:
I was sure that both events were taken from the same place, but it wasn't true! So I added support for AbiV1.Event. Good catch!